### PR TITLE
fix: Apple Sign In

### LINKED ISSUE #154: Crash when Apple Sign in enabled
When I have apple sign in enabled in my .env hasura-auth crashes with error:
```
work-hasura-auth-1  | (node:20) UnhandledPromiseRejectionWarning: Error: The OAuth2 adapter used to refresh the token is not configured correctly. Use the setRefreshOAuth2 option to return a OAuth 2.0 adapter.
work-hasura-auth-1  |     at Object.AuthTokenRefresh.use (/app/node_modules/.pnpm/passport-oauth2-refresh@2.1.0/node_modules/passport-oauth2-refresh/lib/refresh.js:86:11)
work-hasura-auth-1  |     at initProvider (/app/dist/routes/signin/providers/utils.js:161:43)
work-hasura-auth-1  |     at exports.default (/app/dist/routes/signin/providers/apple.js:17:30)
work-hasura-auth-1  |     at Object.<anonymous> (/app/dist/routes/signin/providers/index.js:25:21)
work-hasura-auth-1  |     at Module._compile (internal/modules/cjs/loader.js:1085:14)
work-hasura-auth-1  |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
work-hasura-auth-1  |     at Module.load (internal/modules/cjs/loader.js:950:32)
work-hasura-auth-1  |     at Function.Module._load (internal/modules/cjs/loader.js:790:12)
work-hasura-auth-1  |     at Module.require (internal/modules/cjs/loader.js:974:19)
work-hasura-auth-1  |     at require (internal/modules/cjs/helpers.js:101:18)
work-hasura-auth-1  |     at Object.<anonymous> (/app/dist/routes/signin/index.js:12:37)
work-hasura-auth-1  |     at Module._compile (internal/modules/cjs/loader.js:1085:14)
work-hasura-auth-1  |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
work-hasura-auth-1  |     at Module.load (internal/modules/cjs/loader.js:950:32)
work-hasura-auth-1  |     at Function.Module._load (internal/modules/cjs/loader.js:790:12)
work-hasura-auth-1  |     at Module.require (internal/modules/cjs/loader.js:974:19)
```
.env
```
AUTH_PROVIDER_APPLE_ENABLED=true
AUTH_PROVIDER_APPLE_CLIENT_ID=com....
AUTH_PROVIDER_APPLE_TEAM_ID=2W...
AUTH_PROVIDER_APPLE_KEY_ID=H9...
AUTH_PROVIDER_APPLE_PRIVATE_KEY="PRIVATE_KEY"
AUTH_PROVIDER_APPLE_SCOPE=name,email
```

### PR DESCRIPTION
**Pull request by elitan**
close #154

### CODE REVIEW

File: src/routes/signin/providers/utils.ts [hunk 1]
```diff
@@ -238,15 +239,149 @@ export const initProvider = <T extends Strategy>(
...
-        ...PROVIDERS[strategyName],
-        ...options,
-        callbackURL: `${ENV.AUTH_SERVER_URL}/signin/provider/${strategyName}/callback`,
-        passReqToCallback: true,
-      },
-      manageProviderStrategy(strategyName, transformProfile)
-    );
+    let strategyToUse;
+
+    // Apple is special
```
Comment (plmercereau) [inline src/routes/signin/providers/utils.ts:244]: 😬

**Comment by plmercereau** [comment]
Will it solve #154?

**Comment by Svarto** [comment]
@plmercereau @elitan let me know if I can do anything to help here!

**Comment by plmercereau** [comment]
Can you test it on your end to check if everything runs smoothly? We haven't figured out how to automate Oauth tests, unfortunately.
Cases to test:
- happy path
- with options in the request parameters
- with a `redirectTo` option
- the user cancels/does not accept to continue the authentication

We planned to do it from our end in the upcoming days too, but it's always good to get additional checks and feedback

**Review by plmercereau — APPROVED** [review]

**Pull request merged**